base/utf16.jl

```diff
@@ -136,6 +281,9 @@ function convert(T::Type{UTF16String}, bytes::AbstractArray{UInt8})
     UTF16String(d)
 end
 
+convert(::Type{UTF16String}, str::UTF16String)    = str
```

Comment:
It turns out, that that line in `essentials.jl` never gets used... when I remove the definition above, I get:
```julia
julia> @which convert(UTF16String, utf16("foo"))
convert(::Type{UTF16String}, s::AbstractString) at utf16.jl:75
```
which is defined as:
```julia
convert(::Type{UTF16String}, s::AbstractString) = encode16(s)
```
So I don't think I *can* do anything about those, they are really there for a reason, to keep the AbstractString version from being used.